src/vec.h

```diff
@@ -145,25 +143,31 @@ class Vec : public VecView<T> {
     std::swap(capacity_, other.capacity_);
   }
 
-  inline void push_back(const T &val) {
+  inline void push_back(const T &val, bool seq = false) {
     if (this->size_ >= capacity_) {
       // avoid dangling pointer in case val is a reference of our array
       T val_copy = val;
-      reserve(capacity_ == 0 ? 128 : capacity_ * 2);
+      reserve(capacity_ == 0 ? 128 : capacity_ * 2, seq);
       this->ptr_[this->size_++] = val_copy;
       return;
     }
     this->ptr_[this->size_++] = val;
   }
 
-  void reserve(size_t n) {
+  inline void extend(size_t n, bool seq = false) {
+    if (this->size_ + n >= capacity_)
+      reserve(capacity_ == 0 ? 128 : capacity_ * 2, seq);
```

Comment:
Is this right? If `n` is much larger than `capacity_`, will it actually increase enough?

Reply:
Nice catch, I copied the code from `push_back` and forgot that n can be large.